rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/transform/TbChangeOriginatorNode.java

```diff
@@ -86,13 +99,79 @@ private ListenableFuture<? extends EntityId> getNewOriginator(TbContext ctx, Ent
                 return EntitiesRelatedEntityIdAsyncLoader.findEntityAsync(ctx, original, config.getRelationsQuery());
             case ALARM_ORIGINATOR_SOURCE:
                 return EntitiesAlarmOriginatorIdAsyncLoader.findEntityIdAsync(ctx, original);
+            case ENTITY_SOURCE:
+                return getEntity(ctx, original, msg);
             default:
                 return Futures.immediateFailedFuture(new IllegalStateException("Unexpected originator source " + config.getOriginatorSource()));
         }
     }
 
+    private ListenableFuture<? extends EntityId> getEntity(TbContext ctx, EntityId original, TbMsg msg) {
```

Comment:
Hi @YuriyLytvynchuk, let's move the getEntity logic into a separate class in the same way as AsyncLoaders. So we will be able to use this class in other places as well.